### PR TITLE
Implement Azure deployment and sanity checks

### PR DESCRIPTION
**Pull request by jeniawhite**
### Summary of your changes
Implemented CI for Azure deployment and sanity checks

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/1297

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [x] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [x] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: tests/integration/tests/test_sanity_checks.py
Comment (gurevichdmitry) [inline tests/integration/tests/test_sanity_checks.py:200]: @jeniawhite, the query is currently designed to search based on the `benchmark id` only. However, in your definition of `tests_data` for `cis_azure`, you specified `configuration`, and it appears that this term `{"term": {"resource.type": match_type}}` is not included in the query list.

Comment (jeniawhite) [reply tests/integration/tests/test_sanity_checks.py:200]: That was on purpose due to the fact that our cloud subscription is not static and might change.
In order to remove flakiness or having static resources, we are querying to make sure that we get any findings.
Once we will have a static subscription for the CI, we will introduce checks for every subtype.

Comment (gurevichdmitry) [reply tests/integration/tests/test_sanity_checks.py:200]: Is there a specific task or issue where we can document the absence of these tests for future reference?

**Review by gurevichdmitry — APPROVED** [review]
@jeniawhite Great job, LGTM! 
After merging the code, please execute the `create-environment` workflow on the main branch to verify that everything is functioning as intended.

File: .github/workflows/test-environment.yml
Comment (oren-zohar) [inline .github/workflows/test-environment.yml:1]: Please don't merge until you handle deleting provisioned resources. This also includes adding it to .github/workflows/destroy-environment.yml

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: tests/integration/tests/test_sanity_checks.py
Comment (jeniawhite) [reply tests/integration/tests/test_sanity_checks.py:200]: - https://github.com/elastic/cloudbeat/issues/1521

File: .github/workflows/destroy-environment.yml
Comment (oren-zohar) [inline .github/workflows/destroy-environment.yml:91]: we usually keep longer code blocks as a separate script in the scripts folder

File: .github/workflows/test-environment.yml
Comment (oren-zohar) [inline .github/workflows/test-environment.yml:339]: I feel like we should keep all the cloud env deletion logic under the `delete-cloud-env` just command

File: deploy/test-environments/delete_env.sh
Comment (oren-zohar) [inline deploy/test-environments/delete_env.sh:210]:
```suggestion
                az resource delete --ids "$resource_id" || {
                    echo "Failed to delete resource: $resource_id"
                    continue
                }
```

File: deploy/test-environments/delete_env.sh
Comment (oren-zohar) [inline deploy/test-environments/delete_env.sh:212]:
```suggestion
            az deployment group delete --name "$dep_name" --resource-group "$group" || {
                echo "Failed to delete deployment: $dep_name"
                continue
            }
```

File: deploy/test-environments/delete_env.sh
Comment (oren-zohar) [inline deploy/test-environments/delete_env.sh:222]:
```suggestion
        az group delete --name "$group" --yes || {
            echo "Failed to delete resource group: $group"
            continue
        }
```

**Pull request merged**
